spam/01_spam_tutorial.py

```diff
@@ -192,16 +213,24 @@
 # ### b) Writing an LF to identify spammy comments that use the phrase "check out"
 
 # %% [markdown]
-# Labeling functions in Snorkel are created with the `@labeling_function()` decorator, which wraps a function for evaluating on a single data point (in this case, a row of the DataFrame).
+# Labeling functions in Snorkel are created with the
+# [`@labeling_function()` decorator](https://snorkel.readthedocs.io/en/master/source/snorkel.labeling.lf.html#snorkel.labeling.lf.core.labeling_function),
+# which wraps a function for evaluating on a single data point.
+# Decorators are a convenient Python mechanism for automatically extending the behavior of the decorated function,
+# and you can learn more [here](https://realpython.com/primer-on-python-decorators/).
 #
 # Let's start developing an LF to catch instances of commenters trying to get people to "check out" their channel, video, or website.
 # We'll start by just looking for the exact string `"check out"` in the text, and compare that to looking for just `"check"` in the text.
-# We'll write a Python function over a single data point (here, a [Pandas `Series` object](https://pandas.pydata.org/pandas-docs/stable/reference/series.html)) to express the two versions of our rule, then add the decorator.
+# We'll write a Python function over a single data point to express the two versions of our rule, then add the decorator.
+# Because our data here is stored as a Pandas DataFrame, a single data point `x` will be a [Pandas `Series` object](https://pandas.pydata.org/pandas-docs/stable/reference/series.html).
+# However, these LFs will work for any object with an attribute named `text`.
+# We discuss using other types of data points below when we introduce LF appliers.
 
 # %%
```

Comment:
(Caveat: I don't know what feedback was behind this change).
Nit: A lot of previous cells have been split into multiple cells. Doesn't that make going through the notebook more tedious? (more key presses). Running an import statement more than once isn't bad, so keeping the import in same cell as the function that uses the import doesn't really hurt.

Reply:
People were failing on imports for random reasons and didn't parse the error correctly. This makes it explicit. Can go either way. Reasonable to expect people to understand error messages if the clicking overhead is high